### ISSUE TITLE
Insert mutation with value fails if column preset is set, all permissions correct

### ISSUE THREAD

**Issue by rocketraman** [hasura/graphql-engine#5967]
I'm doing an insert mutation that fails when run with a particular limited role:

```
{
  "errors": [
    {
      "extensions": {
        "path": "$.selectionSet.insert_xs_y_one.args.object.x.data.user_id",
        "code": "validation-failed"
      },
      "message": "field \"user_id\" not found in type: 'xs_insert_input'"
    }
  ]
}
```

with a particular role.

I've verified the permissions -- the role is allowed to set the `user_id` value and the token contains the correct and expected claims.

The mutation should insert into two tables: the mutation object writes to the child table, and this table has an object relationship to a parent table, where the foreign key does include the `user_id` column. I have tried setting the `user_id` value on each of the child and parent objects, as well as both, and all of these cases error.

The problem appears to be a "Column Preset" -- the column preset in this case is set to `X-Hasura-User-Id` which is the same value for user_id that is being passed in the mutation. If the column preset is removed, then the mutation works (but of course no longer works if the user_id is not specified in the mutation). If the column preset is left, then the mutation only works if *no* user_id is specified in the mutation.

I don't believe this should be the case -- the column preset should default a missing value, not prevent a mutation that contains the value from working (and furthermore, the value *matches* the preset).

Hasura v1.3.2

**Comment by rocketraman** [comment]
> Column presets are _hard_ presets and this is by design. Hence, they are removed from the schema as well.

@tirumaraiselvan  If that is true, the documentation is *really* not clear about this. Here is what the section about column presets (https://hasura.io/docs/1.0/graphql/core/schema/default-values/column-presets.html#column-presets) says -- I've bolded elements of this that imply these are *soft* presets, not *hard* presets.

> Let’s say you want certain fields to have their values set automatically **when not explicitly passed** using session variables or fixed values when a new row is created with a particular user role.
>
> Hasura GraphQL engine’s column presets let you define role-based **default values** for any field/column. These values can either be a session variable value or a static value.

The whole section is named "Setting **default** values for fields using role-based column presets". The term "default" implies a soft preset, not a hard preset.

**Comment by tirumaraiselvan** [comment]
That's true. The language can definitely be improved here. cc: @marionschleifer

**Comment by carlpaten** [comment]
FYI, without soft presets, there is no backwards-incompatible path away from presets. In our org we've concluded that we should avoid presets and rely entirely on permissions instead.

**Comment by rocketraman** [comment]
This is still a problem with 2.13.
